pulsar/schema.go

```
@@ -219,6 +226,107 @@ func (ps *ProtoSchema) GetSchemaInfo() *SchemaInfo {
+			Dependency:       fileProtoMap[k].Dependency,
+			PublicDependency: fileProtoMap[k].PublicDependency,
+			MessageType:      fileProtoMap[k].MessageType,
+			EnumType:         fileProtoMap[k].EnumType,
+			Service:          fileProtoMap[k].Service,
+			Extension:        fileProtoMap[k].Extension,
+			Options:          fileProtoMap[k].Options,
+			SourceCodeInfo:   fileProtoMap[k].SourceCodeInfo,
+			Syntax:           fileProtoMap[k].Syntax,
+		}
+		fileDescList = append(fileDescList, &value)
+	}
+	fileDescSet := descriptorpb.FileDescriptorSet{
+		File: fileDescList,
+	}
+	bytesData, err := proto.Marshal(&fileDescSet)
+	if err != nil {
+		log.Fatalf("get serialized proto file data error: %v", err)
+	}
+	schemaData := ProtoNativeSchemaData{
+		FileDescriptorSet:      bytesData,
+		RootMessageTypeName:    string(message.ProtoReflect().Descriptor().FullName()),
+		RootFileDescriptorName: fileDesc.Path(),
+	}
+	jsonData, err := json.Marshal(schemaData)
+	if err != nil {
+		log.Fatalf("get json schema data for proto native schema error: %v", err)
+	}
+	return string(jsonData)
+}
+
+type ProtoNativeSchemaData struct {
+	FileDescriptorSet      []byte `json:"fileDescriptorSet"`
+	RootMessageTypeName    string `json:"rootMessageTypeName"`
+	RootFileDescriptorName string `json:"rootFileDescriptorName"`
+}
+
+func GetFileProto(fileDesc protoreflect.FileDescriptor, protoMap map[string]*descriptorpb.FileDescriptorProto) {
+	if fileDesc.Imports().Len() > 0 {
+		for i := 0; i < fileDesc.Imports().Len(); i++ {
+			GetFileProto(fileDesc.Imports().Get(i).ParentFile(), protoMap)
+		}
+	}
+	unResolvedDependencies := make([]string, 0)
+	for i := 0; i < fileDesc.Imports().Len(); i++ {
+		if _, ok := protoMap[fileDesc.Imports().Get(i).Path()]; ok {
+			if !ok {
+				unResolvedDependencies = append(unResolvedDependencies, fileDesc.Imports().Get(i).Path())
+			}
+		}
+	}
+	if len(unResolvedDependencies) > 0 {
+		log.Fatalf("%v can't resolve dependency [%v].",
```

Could you return an error instead of the log?

---

I remove this check.